kmerexpr/exp_grad_solver.py

```diff
@@ -52,16 +52,30 @@ def update_records(grad,normg0,loss,loss0,x, iter, xs,norm_records,loss_records,
     loss_records.append(loss)
     iteration_counts.append(iter)
 
-def exp_grad_solver(loss_grad,  x_0, lrs=None, tol=10**(-8.0), gtol = 10**(-8.0),  n_iters = 10000, verbose=True,  n = None, Hessinv=False):
+def exp_grad_solver(loss_grad, x_0, lrs=None, tol=10**(-8.0), gtol=10**(-8.0), n_iters=10000, verbose=True, n=None, Hessinv=False, callback=None):
     """
     Exponentiated Gradient Descent for minimizing
     max l(x)  s.t. sum(x) = 1 and x>0
 
     updates take the form of
     x^{t+1}  =  (x^t_i exp( lr_t  grad^t_i )) /(sum (x^t_j exp(lr_t  grad^t_j )).
 
-    n = number of sub functions in l(x) = sum_{i=1}^n l_i(x), Assumes l_i(x), nabla l_i(x) = loss_grad(x, i) 
-    batchsize = number of 
+    Args:
+        loss_grad (Callable): Function to compute the loss and gradient.
+        x0 (NDarray): Initial parameters.
+        lrs: Learning rate. Either a string ("Armijo"), a scalar learning rate,
+            a list of learning rates to use at each step.
+        tol (float): Tolerance for the stopping condition on iterates.
+            Stops if the iterates don't move by more than tol.
+        gtol (float): Tolerance for the stopping condition on the gradient norm.
+            Stops if the gradient is smaller than tol.
```

Comment:
I think this should say L2 norm on the first line and mention that it's not the gradient that's smaller than tol but its norm.

Reply:
Fixed to 
```
        gtol (float, optional): L1 Tolerance for the gradient stopping condition.
            Stop if the gradient is smaller than gtol in L1 norm.
```